### PR TITLE
[SPARK-33596][SS] NPE when there is no watermark metrics

### PR DESCRIPTION
**Pull request by uncleGen**
### What changes were proposed in this pull request?
add check if there is watermark metrics or not

### Why are the changes needed?
NPE when there is no watermark metrics, and then make some UT flaky.

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
Existing UTs.

### CODE REVIEW

**Comment by uncleGen** [comment]
cc @HeartSaVioR

File: sql/core/src/main/scala/org/apache/spark/sql/streaming/ui/StreamingQueryStatisticsPage.scala [hunk 1]
```diff
@@ -150,10 +150,10 @@ private[ui] class StreamingQueryStatisticsPage(parent: StreamingQueryTab)
     if (query.lastProgress.eventTime.containsKey("watermark")) {
       val watermarkData = query.recentProgress.flatMap { p =>
         val batchTimestamp = parseProgressTimestamp(p.timestamp)
-        val watermarkValue = parseProgressTimestamp(p.eventTime.get("watermark"))
```
Comment (uncleGen) [inline sql/core/src/main/scala/org/apache/spark/sql/streaming/ui/StreamingQueryStatisticsPage.scala:153]: when there is not watermark metrics, `p.eventTime.get("watermark")` will return a `null`. NPE will be thrown in `parseProgressTimestamp`.

Comment (xuanyuanking) [reply sql/core/src/main/scala/org/apache/spark/sql/streaming/ui/StreamingQueryStatisticsPage.scala:153]: Could you give more details about the NPE? We already check the `watermark` key in lastProgress.

Comment (HeartSaVioR) [reply sql/core/src/main/scala/org/apache/spark/sql/streaming/ui/StreamingQueryStatisticsPage.scala:153]: Is there a case earlier batch doesn't have "watermark" while later batch does? It'd be nice if you could refer somewhere you encounter this.

**Comment by uncleGen** [comment]
can not reproduce again, close first

File: sql/core/src/main/scala/org/apache/spark/sql/streaming/ui/StreamingQueryStatisticsPage.scala (hunk 1 above)
Comment (xuanyuanking) [reply sql/core/src/main/scala/org/apache/spark/sql/streaming/ui/StreamingQueryStatisticsPage.scala:153]: https://github.com/apache/spark/blob/b665d5881915f042930f502bcc3c6ee3cb00c50d/sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/ProgressReporter.scala#L244-L246

It's possible that lastProgress.eventTime contains `watermark` but recentProgress doesn't, when `hasNewData` = true.

Comment (uncleGen) [reply sql/core/src/main/scala/org/apache/spark/sql/streaming/ui/StreamingQueryStatisticsPage.scala:153]: @HeartSaVioR @xuanyuanking Ignore this false alarm

**Pull request closed without merging**